homeassistant/components/recorder/table_managers/states.py

```diff
@@ -74,6 +90,22 @@ def reset(self) -> None:
         """
         self._last_committed_id.clear()
         self._pending.clear()
+        self._oldest_ts = None
+
+    def load_from_db(self, session: Session) -> None:
+        """Update the cache.
+
+        Must run in the recorder thread.
+        """
+        result = cast(
+            Sequence[Row[Any]],
+            execute_stmt_lambda_element(session, find_oldest_state()),
+        )
+        if not result:
+            ts = None
+        else:
+            ts = result[0].last_updated_ts
+        self._oldest_ts = ts
```

Comment:
I picked the `last_updated_ts` because the `last_changed_ts` is not always set.

Maybe it would be better to use the `last_changed_ts` and fall back to `last_updated_ts` though to avoid some potential edge case where the oldest state is a very long lived states which have been updated.

At the same time, the purge logic considers only the `last_updated_ts`.

Reply:
`last_updated_ts` is likely fine here given thats what the history APIs select on